app/components/lightbox-media.js

```diff
@@ -4,207 +4,172 @@ import { htmlSafe } from '@ember/string';
+				this.nextMedia();
 			}
 		},
 
 		/**
-		 * @param {Event} event
 		 * @returns {void}
 		 */
-		keyDown(event) {
+		swipeRight() {
 			if (this.isGallery) {
-				if (event.keyCode === 39) {
-					// handle right arrow
-					this.nextMedia();
-				} else if (event.keyCode === 37) {
-					// handle left arrow
-					this.prevMedia();
-				}
+				this.prevMedia();
 			}
-
-			this._super(event);
-		},
-
-		/**
-		 * @returns {boolean}
-		 */
-		rightClickHandler() {
-			this.nextMedia();
-			return true;
-		},
-
-		/**
-		 * @returns {boolean}
-		 */
-		leftClickHandler() {
-			this.prevMedia();
-			return true;
 		},
+	},
 
-		/**
-		 * @returns {boolean}
-		 */
-		centerClickHandler() {
-			// Bubble up
-			return false;
-		},
+	setFooter() {},
+	setThumbnails() {},
+	updateGalleryRef() {},
 
-		/**
-		 * @returns {void}
-		 */
-		nextMedia() {
-			this.incrementProperty('currentGalleryRef');
-		},
+	/**
+	 * gets current media or current media from gallery
+	 */
+	currentMedia: computed('model', 'isGallery', 'currentGalleryRef', function () {
+		const current = this.model;
 
-		/**
-		 * @returns {void}
-		 */
-		prevMedia() {
-			this.decrementProperty('currentGalleryRef');
-		},
+		return this.isGallery ? current[this.currentGalleryRef] : current;
+	}),
 
-		/**
-		 * @returns {void}
-		 */
-		updateState() {
-			this.updateHeader();
-			this.updateFooter();
-			this.lightbox.set('file', normalizeToUnderscore(this.get('currentMedia.title')));
+	currentGalleryRef: computed('model.galleryRef', {
+		get() {
+			return this.get('model.galleryRef') || 0;
 		},
 
-		/**
-		 * @returns {void}
-		 */
-		updateHeader() {
-			let header = null;
+		set(key, value) {
+			const galleryLength = this.galleryLength - 1;
 
-			if (this.isGallery) {
-				header = `${(this.currentGalleryRef + 1)} / ${this.galleryLength}`;
+			if (value < 0) {
+				value = galleryLength;
+			} else if (value > galleryLength) {
+				value = 0;
 			}
 
-			this.setHeader(header);
-		},
+			this.updateGalleryRef(value);
 
-		/**
-		 * @returns {void}
-		 */
-		updateFooter() {
-			const currentMedia = this.currentMedia;
-
-			if (currentMedia && currentMedia.caption) {
-				this.setFooter(htmlSafe(currentMedia.caption));
-			} else {
-				this.setFooter(null);
-			}
+			return value;
 		},
+	}),
+
+	galleryLength: computed('isGallery', 'model', function () {
+		return this.isGallery ? this.model.length : -1;
+	}),
+
+	/**
+	 * checks if current displayed media is a gallery
+	 */
+	isGallery: computed('model', function () {
+		return isArray(this.model);
+	}),
+
+	/**
+	 * checks if current media is a video or image and which lightbox component to render
+	 */
+	lightboxComponent: computed('currentMedia', function () {
+		const currentMedia = this.currentMedia;
+
+		return currentMedia && currentMedia.url && currentMedia.type ? `lightbox-${currentMedia.type}` : null;
+	}),
+
+	currentMediaObserver: observer('currentMedia', function () {
+		this.updateState();
```

Comment:
can this be called next to where currentMedia is updated so we can remove observer?

Reply:
it would require bigger refactor of lightbox component